lib/core/diagnostics.js

```diff
@@ -114,6 +120,76 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
   isClientSet = true
 }
 
+// Track fetch requests
+if (fetchDebuglog.enabled) {
+  const debuglog = fetchDebuglog.enabled
+
+  diagnosticsChannel.channel('undici:fetch:start').subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch request has started request to %s',
+      input
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:end').subscribe(evt => {
+    const {
+      input,
+      error
+    } = evt
+    debuglog(
+      'received response from %s, error %s',
+      input,
+      error?.message
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:asyncStart').subscribe(evt => {
+    const {
+      input,
+      result,
+      error
+    } = evt
+    debuglog(
+      'received response for %s - HTTP %d, error is  %s',
+      input,
+      result.status,
+      error?.message
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:asyncStart').subscribe(evt => {
+    const {
+      input,
+      result,
+      error
+    } = evt
+    debuglog(
+      'received response for %s - HTTP %d, error is %s',
+      input,
+      result.status,
+      error?.message
+    )
+  })
```

Comment:
is this right? It is the same as above?

Reply:
It was supposed to be for `asyncEnd` I updated it